lib/SIL/Utils/OwnershipUtils.cpp

```diff
@@ -1137,56 +1144,73 @@ swift::findTransitiveUsesForAddress(SILValue projectedAddress,
-        isa<UncheckedAddrCastInst>(user) || isa<MarkMustCheckInst>(user)) {
+        isa<UncheckedAddrCastInst>(user) || isa<MarkMustCheckInst>(user) ||
+        isa<MarkUninitializedInst>(user) || isa<DropDeinitInst>(user) ||
+        isa<ProjectBlockStorageInst>(user) || isa<UpcastInst>(user) ||
+        isa<TuplePackElementAddrInst>(user)) {
       transitiveResultUses(op);
       continue;
     }
 
     if (auto *builtin = dyn_cast<BuiltinInst>(user)) {
       if (auto kind = builtin->getBuiltinKind()) {
-        if (*kind == BuiltinValueKind::TSanInoutAccess) {
-          leafUse(op);
+        switch (*kind) {
+        case BuiltinValueKind::TSanInoutAccess:
+        case BuiltinValueKind::ResumeThrowingContinuationReturning:
+        case BuiltinValueKind::ResumeNonThrowingContinuationReturning:
+        case BuiltinValueKind::Copy:
+        case BuiltinValueKind::GenericAdd:
+        case BuiltinValueKind::GenericFAdd:
+        case BuiltinValueKind::GenericAnd:
+        case BuiltinValueKind::GenericAShr:
+        case BuiltinValueKind::GenericLShr:
+        case BuiltinValueKind::GenericOr:
+        case BuiltinValueKind::GenericFDiv:
+        case BuiltinValueKind::GenericMul:
+        case BuiltinValueKind::GenericFMul:
+        case BuiltinValueKind::GenericSDiv:
+        case BuiltinValueKind::GenericExactSDiv:
+        case BuiltinValueKind::GenericShl:
+        case BuiltinValueKind::GenericSRem:
+        case BuiltinValueKind::GenericSub:
+        case BuiltinValueKind::GenericFSub:
+        case BuiltinValueKind::GenericUDiv:
+        case BuiltinValueKind::GenericExactUDiv:
+        case BuiltinValueKind::GenericURem:
+        case BuiltinValueKind::GenericFRem:
+        case BuiltinValueKind::GenericXor:
+        case BuiltinValueKind::TaskRunInline:
+          visitor.callVisitUse(op);
           continue;
+        default:
+          break;
         }
       }
     }
 
-    // If we have a load_borrow, add it's end scope to the liveness requirement.
-    if (auto *lbi = dyn_cast<LoadBorrowInst>(user)) {
-      // FIXME: if we can assume complete lifetimes, then this should be
-      // as simple as:
-      //   for (Operand *use : lbi->getUses()) {
-      //     if (use->endsLocalBorrowScope()) {
-      if (!findInnerTransitiveGuaranteedUses(lbi, foundUses)) {
-        result = meet(result, AddressUseKind::PointerEscape);
-      }
-      continue;
-    }
-
     // TODO: Merge this into the full apply site code below.
     if (auto *beginApply = dyn_cast<BeginApplyInst>(user)) {
-      if (foundUses) {
-        // TODO: the empty check should not be needed when dead begin_apply is
-        // disallowed.
-        if (beginApply->getTokenResult()->use_empty()) {
-          leafUse(op);
-        } else {
-          llvm::copy(beginApply->getTokenResult()->getUses(),
-                     std::back_inserter(*foundUses));
+      // TODO: the empty check should not be needed when dead begin_apply is
```

Comment:
Why are `begin_apply` and `load_borrow` handled in different places? They seem like equivalent problems.

Reply:
You are right. I can do a similar treatment to what I did with load_borrow.

Reply:
If I change this, I want to do it in a subsequent commit that isn't so big.

Reply:
I took care of it since I needed to do some more work anyways.